Sources/BasedClient/Client/BasedClient.swift

```diff
@@ -109,10 +109,11 @@ final class BasedClient: BasedClientProtocol {
     var functionCallbacks: FunctionCallbackStore
     
     var basedCClient: BasedCClientProtocol
-    
+    var enableTls: Bool = false
```

Comment:
when using the client in an on-prem scenario, we would need to set this to true?